subsys/ieee802154/ieee802154_nrf5_cfg.c

```diff
@@ -0,0 +1,146 @@
+		.active_high  = 1,
+		.gpio_pin     = CONFIG_IEEE802154_NRF5_FEM_PA_PIN,
+		.gpiote_ch_id = CONFIG_IEEE802154_NRF5_FEM_PA_GPIOTE_CHANNEL
+	},
+	.lna_pin_config =
+	{
+		.enable       = CONFIG_IEEE802154_NRF5_FEM_LNA_PIN_CTRL_ENABLE,
+		.active_high  = 1,
+		.gpio_pin     = CONFIG_IEEE802154_NRF5_FEM_LNA_PIN,
+		.gpiote_ch_id = CONFIG_IEEE802154_NRF5_FEM_LNA_GPIOTE_CHANNEL
+	},
+#if CONFIG_NRF_802154_FEM_TYPE_3_PIN
+	.pdn_pin_config =
+	{
+		.enable       = CONFIG_IEEE802154_NRF5_FEM_PDN_PIN_CTRL_ENABLE,
+		.active_high  = 1,
+		.gpio_pin     = CONFIG_IEEE802154_NRF5_FEM_PDN_PIN,
+		.gpiote_ch_id = CONFIG_IEEE802154_NRF5_FEM_PDN_GPIOTE_CHANNEL
+	},
+	.ppi_ch_id_pdn = CONFIG_IEEE802154_NRF5_FEM_PDN_PPI_CHANNEL,
+#endif
+	.ppi_ch_id_set = CONFIG_IEEE802154_NRF5_FEM_SET_PPI_CHANNEL,
+	.ppi_ch_id_clr = CONFIG_IEEE802154_NRF5_FEM_CLR_PPI_CHANNEL,
+};
+#endif
+
+static void nrf5_gpio_configure(void)
+{
+#if CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_MODE_MANUAL || IEEE802154_NRF5_ANT_DIVERSITY_MODE_AUTO
+	nrf_gpio_cfg_output(CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_PIN);
+	nrf_gpio_pin_clear(CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_PIN); // Use antenna at "ANT1" port by default
+#endif
+
+#if CONFIG_IEEE802154_NRF5_FEM_PRESENT
+	// Set default FEM pin direction.
+	nrf_gpio_cfg_output(CONFIG_IEEE802154_NRF5_FEM_PA_PIN);
+	nrf_gpio_cfg_output(CONFIG_IEEE802154_NRF5_FEM_LNA_PIN);
+	nrf_gpio_cfg_output(CONFIG_IEEE802154_NRF5_FEM_PDN_PIN);
+	nrf_gpio_cfg_output(CONFIG_IEEE802154_NRF5_FEM_MODE_PIN);
+
+	nrf_gpio_cfg_output(CONFIG_IEEE802154_NRF5_FEM_MOSI_PIN);
+	nrf_gpio_cfg_default(CONFIG_IEEE802154_NRF5_FEM_MISO_PIN);
+	nrf_gpio_cfg_output(CONFIG_IEEE802154_NRF5_FEM_CLK_PIN);
+	nrf_gpio_cfg_output(CONFIG_IEEE802154_NRF5_FEM_CSN_PIN);
+
+	// Set default FEM pin polarity.
+	nrf_gpio_pin_clear(CONFIG_IEEE802154_NRF5_FEM_PA_PIN);     // Disable PA, radio driver will override this setting
+	nrf_gpio_pin_clear(CONFIG_IEEE802154_NRF5_FEM_LNA_PIN);    // Disable LNA, radio driver will override this setting
+	nrf_gpio_pin_clear(CONFIG_IEEE802154_NRF5_FEM_PDN_PIN);    // Disable FEM, radio driver will override this setting
+	nrf_gpio_pin_clear(CONFIG_IEEE802154_NRF5_FEM_MODE_PIN);   // Use POUTA_PROD TX gain by default
+
+	nrf_gpio_pin_set(CONFIG_IEEE802154_NRF5_FEM_MOSI_PIN);     // SPI mode not used. Use high polarity by default.
+	nrf_gpio_pin_set(CONFIG_IEEE802154_NRF5_FEM_CLK_PIN);      // SPI mode not used. Use high polarity by default.
+	nrf_gpio_pin_clear(CONFIG_IEEE802154_NRF5_FEM_CSN_PIN);    // SPI mode not used. Use low polarity by default.
+#endif
+}
+
+static int nrf5_cfg(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	nrf5_gpio_configure();
+
+#if CONFIG_IEEE802154_NRF5_FEM_PRESENT
+	if (!nrf_fem_interface_configuration_set(&fem_iface_config)) {
+		LOG_ERR("Failed to configure FEM");
+	}
+#endif
+
+#if CONFIG_IEEE802154_NRF5_ANT_DIVERSITY_MODE != IEEE802154_NRF5_ANT_DIVERSITY_MODE_DISABLED
+	nrf_802154_ant_diversity_config_t cfg = nrf_802154_antenna_config_get();
```

Comment:
You need an empty line after variable definition.

Reply:
fixed